src/common/input/handler.rs

```diff
@@ -276,34 +282,160 @@ impl InputHandler {
     }
 }
 
+lazy_static! {
```

Comment:
It still looks like `lazy_static!` is the best we can do with `HashMap`s, but does also seem a touch excessive for a constant. Do you reckon a `const` array or tuple array would work? There seems to be [a decent amount to work with](https://doc.rust-lang.org/reference/const_eval.html) nowadays, but let me know if you think it's worth it!

Reply:
Rewrite this with a const fn.
It looks a bit dirty because const fn does not support `for`, `PartialEq`, etc.